packages/iov-keycontrol/src/keyring-entries/secp256k1hdwallet.spec.ts

```diff
@@ -0,0 +1,78 @@
+import { PrehashType, SignableBytes } from "@iov/bcp-types";
+import { Secp256k1, Slip10RawIndex } from "@iov/crypto";
+import { Encoding } from "@iov/encoding";
+import { ChainId } from "@iov/tendermint-types";
+
+import { Secp256k1HdWallet } from "./secp256k1hdwallet";
+
+const { fromHex, toAscii } = Encoding;
+
+describe("Secp256k1HdWallet", () => {
+  it("returns the concrete type when creating from entropy", () => {
+    const wallet = Secp256k1HdWallet.fromEntropy(fromHex("51385c41df88cbe7c579e99de04259b1aa264d8e2416f1885228a4d069629fad"));
+    expect(wallet).toEqual(jasmine.any(Secp256k1HdWallet));
+    expect(wallet.implementationId).toEqual("secp256k1-hd");
+  });
+
+  it("returns the concrete type when creating from mnemonic", () => {
+    const wallet = Secp256k1HdWallet.fromMnemonic("execute wheel pupil bachelor crystal short domain faculty shrimp focus swap hazard");
+    expect(wallet).toEqual(jasmine.any(Secp256k1HdWallet));
```

Comment:
nice check... this is like `instanceOf` in curried form, right?

Reply:
jap